recipes-core/images/nilrt-safemode-initramfs.bb

```diff
@@ -23,8 +24,6 @@ IMAGE_INSTALL_NODEPS += "\
 	${NI_PROPRIETARY_SAFEMODE_PACKAGES} \
 "
 
-BAD_RECOMMENDATIONS_append_pn-${PN} += "shared-mime-info"
```

Comment:
This initramfs does not exclude `ca-certificates`, but it looks like with your change, it will. Is that intended? Not sure why we exclude the certificates from the runmode initramfs.

Reply:
[Ionel thought they were inappropriate]([https://github.com/ni/meta-nilrt/commit/243fdb0528ae3559e39dfac27cd29ad0f2291a7d#diff-99cc1ba30ac7f8cf446cf4217a17416d1d3a6e3c24d2911a7f424a42bf2017f0). :shrug: It's a `BAD_RECOMMENDS`, so it won't block opkg from installing `ca-certificates` if a requested package actually depends on it.

```
(safemode) admin@NI-cRIO-903x-VM-a18e5030:~# opkg whatdepends -A ca-certificates
Root set:
  ca-certificates
What depends on root set
        ca-certificates-dev 20210119-r0.75      depends on ca-certificates (= 20210119-r0.75)
        docker-ce v19.03.15-ce+git99e3ed89195c4e551e87aad1e7453b65456b03ad-r0.0 depends on ca-certificates
        docker-ce-dev v19.03.15-ce+git99e3ed89195c4e551e87aad1e7453b65456b03ad-r0.0     depends on docker-ce (= v19.03.15-ce+git99e3ed89195c4e551e87aad1e7453b65456b03ad-r0.0)
        ni-ca-certs 21.3.0.49152-0+f0   depends on ca-certificates
(safemode) admin@NI-cRIO-903x-VM-a18e5030:~# opkg whatrecommends -A ca-certificates
Root set:
  ca-certificates
What recommends root set
        wget 1.21.1-r0.0        recommends ca-certificates
        python3-crypt 3.9.5-r0.9        recommends ca-certificates
        libcurl4 7.75.0-r0.20   recommends ca-certificates
```

It seems like it gets installed to safemode anyway via the `ni-ca-certs` dependency.

Reply:
I'm guessing Ionel excluded them because it is non-trivial to update/revoke a certificate when it's built-in the initramfs i.e. you can't do it by just updating a package via opkg.